awesome_tshirt/static/src/dashboard.js

```diff
@@ -1,11 +1,83 @@
 /** @odoo-module **/
 
-import { Component } from "@odoo/owl";
+
+import { useService } from "@web/core/utils/hooks";
+import { getDefaultConfig } from "@web/views/view";
+import { Component, useSubEnv } from "@odoo/owl";
+import { Domain } from "@web/core/domain";
 import { registry } from "@web/core/registry";
+import { Layout } from "@web/search/layout";
+import { onWillStart } from "@odoo/owl";
+
+import { Card } from "./card/card"
+import { PieChart } from "./pie_chart/pie_chart"
+
+
+class AwesomeDashboard extends Component {
+    statistic_key_to_title = {
+        "average_quantity": this.env._t("Average amount of t-shirt by order this month"),
+        "average_time": this.env._t("Average time for an order to go from 'new' to 'cancelled'"),
+        "nb_cancelled_orders": this.env._t("Number of cancelled orders this month"),
+        "nb_new_orders": this.env._t("Number of new orders this month"),
+        "total_amount": this.env._t("Total amount of new orders this month"),
+    }
+
+    setup() {
+        const tshirtStatisticsService = useService("tshirtStatisticsService");
+        this.action = useService("action");
+
+        onWillStart(async () => {
+            this.statistics = await tshirtStatisticsService.get()
+        });
+
+        useSubEnv({
+            config: {
+                ...getDefaultConfig(),
+                ...this.env.config,
+            },
+        });
+    }
+
+    #openOrders(name, domain) {
```

Comment:
In the Odoo JavaScript codebase we don't use private class methods:
-  If you use private methods and you wrap your object into a proxy (with `reactive` or `useState`) you'll have a hard time since the `this` will be binded to the proxy and not the object itself. It add a layer of complexity in the code.

example:
```js
class MySuperModel {

    constructor(birthday) {
        this.birthday = birthday;
        this.age = 2023 - birthday;
    }
    setBirthday(birtdhay) {
        this.birthday = birtdhay;
        this.#mySuperPrivateMethod();
    }
    #mySuperPrivateMethod() {
        this.age = 2023 - this.birthday;
        console.log("hello world !");
    }
}

export class ClickerSystray extends Component {
    static template = "awesome_clicker.ClickerSystray";
    static components = { ClickerValue, Dropdown, DropdownItem };
    
    setup() {
        ...
        const model = new MySuperModel(2010);
        model.setBirthday(2009); // OK
        const trackedModel = useState(new MySuperModel(2008));
        trackedModel.setBirthday(2008); // NOK, 'this' is binded to the proxy and not the object -> error
        ...
    }
}
```
- Philosophically speaking, in Odoo we like to make our code extensible (for partners, PS Tech team, ...). Private methods cannot be monkey patched therefore partners will have a hard time to customize this (imagine they want to modify the action).

Conventionally we put an underscore to make a method private. It is not enforced by the interpreter but at least partners can do the crap they want to create their features